lib/client.js

```diff
@@ -104,6 +110,19 @@ function messageHandler(client, event) {
   }
 }
 
+// When doing singular operations and we retrieve and error this function will throw that error.
+function throwUnlessBulk(path, result) {
+  var isSingularOperation = typeof path === 'string';
+
+  // CRUFT: When framework starts always appending errors we can remove the extra results.errors check
+  if(isSingularOperation && result.errors) {
+    var err = result.errors.find(function(error) { return error.path === path; });
```

Comment:
Array.prototype.find not supported in all browsers...
Also I think this should be an object while results is still an object...